core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -363,6 +364,7 @@ public void testSegments() throws Exception {
             assertThat(segments.get(1).isCompound(), equalTo(false));
 
             engine.config().setCompoundOnFlush(true);
+            engine.onSettingsChanged();
```

Comment:
OK again :) I misread "we push settings changes down to IW only during flush and refresh" to mean that Lucene does it that way. I was wrong.